ignite/engine/engine.py

```diff
@@ -673,6 +678,7 @@ def switch_batch(engine):
                         )
                     )
                 self.state.max_epochs = max_epochs
+                self.state.max_iters = max_epochs * self.state.epoch_length
```

Comment:
I think this can lead to an exception if user stopped and try to rerun the engine on an iterable without specifying `epoch_length`.